modular_skyrat/modules/microfusion/code/microfusion_gun_attachments.dm

```diff
@@ -0,0 +1,392 @@
+	. = ..()
+	if(toggle)
+		microfusion_gun.heat_dissipation_bonus -= cooling_rate_increase
+	microfusion_gun.fire_sound = microfusion_gun.chambered?.fire_sound
+
+/*
+RGB ATTACHMENT
+
+Enables you to change the light color of the laser.
+*/
+/obj/item/microfusion_gun_attachment/rgb
+	name = "phase emitter spectrograph"
+	desc = "Enables the phase emitter to change it's output color."
+	icon_state = "attachment_rgb"
+	attachment_overlay_icon_state = "attachment_rgb"
+	/// What color are we changing the sprite to?
+	var/color_to_apply = COLOR_MOSTLY_PURE_RED
+
+/obj/item/microfusion_gun_attachment/rgb/process_fire(obj/item/gun/microfusion/microfusion_gun, obj/item/ammo_casing/chambered)
+	. = ..()
+	chambered?.loaded_projectile.icon_state = "laser_greyscale"
+	chambered?.loaded_projectile.color = color_to_apply
+	chambered?.loaded_projectile.light_color = color_to_apply
+
+/obj/item/microfusion_gun_attachment/rgb/proc/select_color(mob/living/user)
+	var/new_color = input(user, "Please select your new projectile color", "Laser color", color_to_apply) as null|color
+
+	if(!new_color)
+		return
+
+	color_to_apply = new_color
+
+/obj/item/microfusion_gun_attachment/rgb/attack_self(mob/user, modifiers)
+	. = ..()
+	select_color(user)
+
+/obj/item/microfusion_gun_attachment/rgb/get_modify_data()
+	return list(list("title" = "Change Color", "icon" = "wrench", "reference" = "color", "color" = "blue"))
+
+/obj/item/microfusion_gun_attachment/rgb/run_modify_data(params, mob/living/user)
+	if(params == "color")
+		select_color(user)
+
+/*
+RAIL ATTACHMENT
+
+Allows for flashlights bayonets and adds 1 slot to equipment.
+*/
+/obj/item/microfusion_gun_attachment/rail
+	name = "gun rail attachment"
+	desc = "A simple set of rails that attaches to weapon hardpoints. Allows for 3 more attachment slots and the instillation of a flashlight or bayonet."
+	icon_state = "attachment_rail"
+	attachment_overlay_icon_state = "attachment_rail"
+	slot = GUN_SLOT_RAIL
+
+/obj/item/microfusion_gun_attachment/rail/run_attachment(obj/item/gun/microfusion/microfusion_gun)
+	. = ..()
+	microfusion_gun.can_flashlight = TRUE
+	microfusion_gun.can_bayonet = TRUE
+
+/obj/item/microfusion_gun_attachment/rail/remove_attachment(obj/item/gun/microfusion/microfusion_gun)
+	. = ..()
+	microfusion_gun.gun_light = initial(microfusion_gun.can_flashlight)
+	if(microfusion_gun.gun_light)
+		microfusion_gun.gun_light.forceMove(get_turf(microfusion_gun))
+		microfusion_gun.clear_gunlight()
+	microfusion_gun.can_bayonet = initial(microfusion_gun.can_bayonet)
+	if(microfusion_gun.bayonet)
+		microfusion_gun.bayonet.forceMove(get_turf(microfusion_gun))
+		microfusion_gun.clear_bayonet()
+	microfusion_gun.remove_all_attachments()
+
+/*
+SCOPE ATTACHMENT
+
+Allows for a scope to be attached to the gun.
+DANGER: SNOWFLAKE ZONE
+*/
+/obj/item/microfusion_gun_attachment/scope
+	name = "scope attachment"
+	desc = "A simple scope that allows for a zoom level."
```

Comment:
desc = "A simple telescopic scope, allowing for long-ranged use of the weapon. However, these do not provide any night vision."